Automattic/wp-calypso
# Fix/cleanup yarn lock

#### Changes proposed in this Pull Request

* Adds CI step to Lint job to validate that `yarn.lock` does not have uncommitted changes.
* Cleans up existing uncommitted changes in `yarn.lock`
* Removed `--frozen-lockfile` since it doesn't do what we expect (yarnpkg/yarn#3313 yarnpkg/yarn#5847) to avoid any future confusion

From https://app.circleci.com/pipelines/github/Automattic/wp-calypso/58032/workflows/819cbf5a-ec40-4989-a89a-c57186712b9d/jobs/715945/steps
![image](https://user-images.githubusercontent.com/975703/81145174-0e79d600-8f76-11ea-9025-4cc3ac5ceaf4.png)


#### Testing instructions

1. Checkout the branch, delete `yarn.lock` and run `yarn`. Run this script in the console and verify it shows the error:

```
DIRTY_FILES=$(git status --porcelain 2>/dev/null)
if [[ ! -z "$DIRTY_FILES" ]]; then
  echo "Repository contains uncommitted changes: "
  echo "$DIRTY_FILES"
  echo "You may need to checkout the branch, run 'yarn' and commit those files."
fi
```

2. Revert local changes to `yarn.lock` and run the script again, you shouldn't see any error.
